script.js

```diff
@@ -1,4 +1,577 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+// More Comfortable
+// Include base, score, leaderboard, lowest combined number mode, auto-generate combined number, variable number of dice features
+
+// GLOBAL VARIABLES
+const CHOOSE_DICECOUNT = "CHOOSE_DICECOUNT";
+const ROLL = "ROLL";
+const AUTO_GENERATE_NUMBER = "AUTO_GENERATE_NUMBER";
+const LEADERBOARD = "LEADERBOARD";
+const HIGHEST_COMBINED_NUMBER_MODE = "HIGHEST_COMBINED_NUMBER_MODE";
+const LOWEST_COMBINED_NUMBER_MODE = "LOWEST_COMBINED_NUMBER_MODE";
+let programState = CHOOSE_DICECOUNT;
+let gameMode = "";
+let diceCount = 0;
+let currentPlayerDiceNumbers = [];
+let currentPlayer = 1;
+let player1FinalNumEachRound = [];
+let player2FinalNumEachRound = [];
+
+const rollOneDice = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+const rollMultipleDice = function (diceCount) {
+  let rollStateMessage = `Player ${currentPlayer}'s turn: <br>`;
+  for (i = 0; i < diceCount; i += 1) {
+    currentPlayerDiceNumbers.push(rollOneDice());
+    let diceId = i + 1;
+    rollStateMessage += `You rolled ${currentPlayerDiceNumbers[i]} for Dice ${diceId}. <br>`;
+  }
+  if (gameMode === HIGHEST_COMBINED_NUMBER_MODE) {
+    rollStateMessage += `Computer is about to automatically concatenate the ${diceCount} digits rolled to create the <b> largest </b> possible number. Please hit the "Submit" button to continue.`;
+  } else if (gameMode === LOWEST_COMBINED_NUMBER_MODE) {
+    rollStateMessage += `Computer is about to automatically concatenate the ${diceCount} digits rolled to create the <b> smallest </b> possible number. Please hit the "Submit" button to continue.`;
+  }
+  return rollStateMessage;
+};
+
+const autoGenerateOptimalCombinedNumber = function () {
+  // MUST initialize this as an empty string or else the value will be undefined during reassignment later
+  let currentPlayerFinalNum = "";
+
+  // Arrange digits from big to small in the array, convert each of them to string and keep them in currentPlayerFinalNum variable
+  if (gameMode === HIGHEST_COMBINED_NUMBER_MODE) {
```

Comment:
Good effort using strict equivalence here, let me know if there's anything that you might want to clarify about this.